applications/ShallowWaterApplication/custom_utilities/estimate_dt_utility.h

```diff
@@ -0,0 +1,243 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+
+#ifndef KRATOS_ESTIMATE_DT_SHALLOW_H_INCLUDED
+#define KRATOS_ESTIMATE_DT_SHALLOW_H_INCLUDED
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "includes/model_part.h"
+#include "includes/kratos_parameters.h"
+
+
+namespace Kratos
+{
+///@addtogroup ShallowWaterApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Utility to estimate the time step in terms of the courant number.
+/** The velocity can be the sum of the convective velocity and the wave speed
+*/
+class KRATOS_API(SHALLOW_WATER_APPLICATION) EstimateDtShallow
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of EstimateDtShallow
+    KRATOS_CLASS_POINTER_DEFINITION(EstimateDtShallow);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+    EstimateDtShallow(ModelPart& rThisModelPart, Parameters& rThisParameters);
+
+    /// Destructor.
+    ~EstimateDtShallow(){}
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    double EstimateDt();
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const;
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const;
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const;
+
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ModelPart& mrModelPart;
```

Comment:
```suggestion
    const ModelPart& mrModelPart;
```
you could try this, I think it could work

Reply:
thanks for your suggestions!